### ISSUE TITLE
SignalListItemFactory doesn't provide object argument to handler

### ISSUE THREAD

**Issue by Nyeksenn** [jwharm/java-gi#161]
I am trying to bind a callback function to the setup and bind signal emitted by SignalListItemFactory.
I have declared a signal in the ui definition as following:

```xml
<object class="GtkSignalListItemFactory">
  <signal name="setup" handler="lstSetupWidget" />
  <signal name="bind" handler="lstBindCategoryName" />
</object>
```
The corresponding handlers look like this:

```java

    @GtkCallback(name = "lstSetupWidget")
    public void lstSetupWidget(GObject object) {
        var listItem = (ListItem) object;
        var label = new Label("");
        label.setMarginTop(12);
        label.setMarginBottom(12);
        listItem.setChild(label);
    }

    @GtkCallback(name = "lstBindCategoryName")
    public void lstBindCategoryName(GObject object) {
        var listItem = (ListItem) object;
        var label = (Label) listItem.getChild();
        var result = (EvaluationResult) listItem.getItem();
        label.setLabel(result.getCategoryName());
    }

```

When i run the program I receive the following error:
```bash
(java:140426): java-gi-CRITICAL **: 15:42:34.327: Cannot invoke method lstSetupWidget in class MetacheckerWindow: wrong number of arguments: 0 expected: 1
(java:140426): java-gi-CRITICAL **: 15:42:34.327: Cannot invoke method lstBindCategoryName in class MetacheckerWindow: wrong number of arguments: 0 expected: 1
```

It seems like when the signal is emitted, that the ListItem is not passed as an argument.

I have looked at an example of a Vala app that does it exactly like that, so I don't think this is a GTK issue.

**Comment by jwharm** [comment]
This is a bug, thanks for reporting it.
Can you attach a small, easily runnable example that I can test with? I don't know how to work with a ListItemFactory from a ui file.

**Comment by jwharm** [comment]
Note to self: This needs to be fixed in the BuilderJavaScope class.

**Comment by jwharm** [comment]
Thanks a lot for the PoC.
The bug has been fixed in the main branch, and I published a new 0.11.1-SNAPSHOT release.

(To use snapshot releases, add https://s01.oss.sonatype.org/content/repositories/snapshots to the repositories in your build script.)
